php/commands/core.php

```diff
@@ -143,6 +151,10 @@ public function download( $args, $assoc_args ) {
 			unlink($temp);
 		}
 
+		if ( version_compare( $from_version, '4.4', '>=' ) && version_compare( $assoc_args['version'], '4.4', '<' ) ) {
```

Comment:
> Currently if either of the checksum calls fail no files will be removed. Is this a reasonable fallback?

I think so, although we should `WP_CLI::warn()` when it happens so the user knows to go clean up.